From aba69c83ca46683a51483767933164d980a2cd57 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 27 Nov 2014 16:33:37 -0500
Subject: [scsi] megaraid_sas: corrected return of wait_event from abort frame path

Message-id: <1417106019-4286-6-git-send-email-thenzl@redhat.com>
Patchwork-id: 100998
O-Subject: [PATCH 5/7] megaraid_sas: corrected return of wait_event from abort frame path
Bugzilla: 1162645
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

Corrected wait_event() call which was waiting for wrong completion
status (0xFF).

Signed-off-by: Sumit Saxena <sumit.saxena@avagotech.com>
Signed-off-by: Kashyap Desai <kashyap.desai@avagotech.com>
Reviewed-by: Tomas Henzl <thenzl@redhat.com>
Cc: <stable@vger.kernel.org>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 170c238701ec38b1829321b17c70671c101bac55)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas_base.c b/drivers/scsi/megaraid/megaraid_sas_base.c
index 53aeed0..4345d93 100644
--- a/drivers/scsi/megaraid/megaraid_sas_base.c
+++ b/drivers/scsi/megaraid/megaraid_sas_base.c
@@ -1007,7 +1007,7 @@ megasas_issue_blocked_abort_cmd(struct megasas_instance *instance,
   cpu_to_le32(upper_32_bits(cmd_to_abort->frame_phys_addr));
 
  cmd->sync_cmd = 1;
- cmd->cmd_status = 0xFF;
+ cmd->cmd_status = ENODATA;
 
  instance->instancet->issue_dcmd(instance, cmd);
 
-- 
1.7.1